sympy/polys/rings.py

```diff
@@ -1066,13 +1072,13 @@ def __rfloordiv__(self, other):
         else:
             return other._floordiv(self)
 
-    def __truediv__(self, other):
+    def __truediv__(self, other: PolyElement[Er] | Er | int) -> PolyElement[Er]:
         ring = self.ring
 
         if not other:
             raise ZeroDivisionError("polynomial division")
         elif ring.is_element(other):
-            return self._truediv(other)
+            return self._truediv(cast('Er', other))
```

Comment:
This cast should not be needed and it casting to the wrong type. It should be `PolyElement[Er]`